packages/dataviews/src/dataform-controls/boolean.tsx

```diff
@@ -17,38 +16,14 @@ export default function Boolean< Item >( {
 	onChange,
 	data,
 	hideLabelFromVision,
+	validity,
 }: DataFormControlProps< Item > ) {
 	const { id, getValue, label } = field;
-	const [ customValidity, setCustomValidity ] =
-		useState<
-			React.ComponentProps<
-				typeof ValidatedToggleControl
-			>[ 'customValidity' ]
-		>( undefined );
 
 	return (
 		<ValidatedToggleControl
 			required={ !! field.isValid.required }
-			onValidate={ ( newValue: any ) => {
```

Comment:
Feedback for @mirka 

onValidate is a useless prop, it's meant to handle custom validation and then we pass the result back as a customValidity prop. The reality is that the same callback can just be triggered in onChange. Removing the prop to avoid race condition is best.